x/gamm/types/msg_swap_sybil.go

```diff
@@ -0,0 +1,44 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Sybil Resistant Fee Swap Msg defines a simple interface for getting
+// - poolIds on a swap msg route
+// - token to apply the sybil resistance to
+type SybilResistantFeeSwap interface {
+	SwapMsgRoute
+
+	PoolIdOnPath() []uint64
+	GetTokenToFee() sdk.Coin
+}
+
+var (
+	_ SybilResistantFeeSwap = MsgSwapExactAmountOut{}
+	_ SybilResistantFeeSwap = MsgSwapExactAmountIn{}
+)
```

Comment:
Should this be expanded to include Join/Exit swap msgs?

Reply:
Changes would be
https://github.com/osmosis-labs/osmosis/blob/6f00aa702e2e72ed20ab82cba87256b7a043307e/x/gamm/types/msg_swap_sybil.go#L10-L20
https://github.com/osmosis-labs/osmosis/blob/6f00aa702e2e72ed20ab82cba87256b7a043307e/x/gamm/types/msg_swap_sybil.go#L46-L58